src/librbd/api/Image.cc

```diff
@@ -345,8 +345,8 @@ int Image<I>::list_children(I *ictx,
         return r;
       }
 
-      for (auto& it : image_names_to_ids) {
-        child_image_id_to_info[it.second] = {it.first, false};
+      for (auto& [name, id] : image_names_to_ids) {
+        child_image_id_to_info.emplace(id, make_pair(name, false));
```

Comment:
Are you saying that replacing ```make_pair(...)``` with ```{...}``` fails to link?

Reply:
@dillaman i think it's 
```c++
child_image_id_to_info[it.second] = {it.first, false};
```
that fails to link.

Reply:
Ok, but what about ```child_image_id_to_info.emplace(id, {name, false})```?

Reply:
but following code fails to compile
```c++
#include <map>
#include <string>

using namespace std;

int main()
{
  std::map<std::string, std::pair<std::string, bool>> child_image_id_to_info;
  string id, name;
  // child_image_id_to_info.emplace(id, make_pair(name, false));
  child_image_id_to_info.emplace(id, {name, false});
}
```
gcc:
```
test.cc: In function ‘int main()’:
test.cc:11:51: error: no matching function for call to ‘std::map<std::__cxx11::basic_string<char>, std::pair<std::__cxx11::basic_string<char>, bool> >::emplace(std::__cxx11::string&, <brace-enclosed initializer list>)’
   child_image_id_to_info.emplace(id, {name, false});
                                                   ^
In file included from /usr/include/c++/7/map:61:0,
                 from test.cc:2:
/usr/include/c++/7/bits/stl_map.h:568:2: note: candidate: std::pair<typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator, bool> std::map<_Key, _Tp, _Compare, _Alloc>::emplace(_Args&& ...) [with _Args = {}; _Key = std::__cxx11::basic_string<char>; _Tp = std::pair<std::__cxx11::basic_string<char>, bool>; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::pair<std::__cxx11::basic_string<char>, bool> > >; typename std::_Rb_tree<_Key, std::pair<const _Key, _Tp>, std::_Select1st<std::pair<const _Key, _Tp> >, _Compare, typename __gnu_cxx::__alloc_traits<_Allocator>::rebind<std::pair<const _Key, _Tp> >::other>::iterator = std::_Rb_tree_iterator<std::pair<const std::__cxx11::basic_string<char>, std::pair<std::__cxx11::basic_string<char>, bool> > >]
  emplace(_Args&&... __args)
  ^~~~~~~
/usr/include/c++/7/bits/stl_map.h:568:2: note:   candidate expects 0 arguments, 2 provided
```

clang:
```
clang++ -std=c++17 -O0 -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib -I/usr/local/opt/llvm/include -g test.cc -lm -o test
test.cc:11:26: error: no matching member function for call to 'emplace'
  child_image_id_to_info.emplace(id, {name, false});
  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/Library/Developer/CommandLineTools/usr/include/c++/v1/map:1040:26: note: candidate
      template ignored: substitution failure : deduced incomplete pack
      <std::__1::basic_string<char> &, (no value)> for template parameter '_Args'
    pair<iterator, bool> emplace(_Args&& ...__args) {
                         ^
1 error generated.
```

Reply:
OK -- how about ```child_image_id_to_info.insert({id, {name, false}})```? That works for me under gcc and clang.